internal/cmdrender/runtime/container.go

```diff
@@ -41,6 +44,34 @@ type ContainerFnPermission struct {
 	AllowNetwork bool
 }
 
+// ContainerFnWrapper wraps the real function filter, prints
+// the function running progress and failures.
+type ContainerFnWrapper struct {
+	Fn *ContainerFn
+}
+
+func (fw *ContainerFnWrapper) Run(r io.Reader, w io.Writer) error {
+	pr := printer.FromContextOrDie(fw.Fn.Ctx)
+	printOpt := printer.NewOpt().Indent(printer.FnIndentation)
+	pr.OptPrintf(printOpt, "[RUNNING] %q\n", fw.Fn.Image)
```

Comment:
Ditto: Previous comment about not mixing presentation and low-low logic doesn't seem to be addressed.

Reply:
Do you mean we should have separate functions for these print statements?

Reply:
is eval using the same presentation? If so, are you duplicating the presentation twice given this package is specific to render? I think you want to extract these out.

Reply:
Eval can use this wrapper in this package for presentation as well with a little tweak. So it's not duplicate.

Reply:
So `eval` would depend on a file in `cmdrender` which is specific to the render command? Or are you extracting this as a common package somewhere else? To avoid back and forth, please hash the plan.

Reply:
IMO eventually eval and render should share the same container runtime. So we should extract the runtime to a separate package so it's not specific to render. For now, yes eval needs to depend on the render runtime.

Plan:
1. Eval depends on this struct in `cmdrender`.
2. Extract render runtime to separate package. So eval will depend on that package.
2. Add functionality to runtime package so eval and render can share the same runtime.

Reply:
https://github.com/GoogleContainerTools/kpt/issues/1849

Reply:
This is technical debt that we have accumulated for now, I can help with the refactor here.